UKHomeOffice/cop-ui
# COP-8097 Add 'you do not have an account page'

Ticket: https://support.cop.homeoffice.gov.uk/browse/COP-8583

# AC

- When a user attempts to log into COP, the system will determine if a valid COP account exists for the user credentials provided. i.e. Active flag in Keycloak = Yes (user has an account), Active flag in Keycloak = No or Null (user does not have an account)

- If a user does not have a valid COP account:
-- they will be prompted to confirm that they want to create one
-- they will then be directed to complete the create COP account form: 
-- they will not be able to access COP until a account creation is complete

- If a user has selected log on on the product page and has a valid account they will be routed to the COP dashboard. The create account form will not be displayed. 

- If a user has selected Create account on the product page and has a valid account they will be displayed an error page before being routed to the COP dashboard. The create account form will not be displayed. 

- If a user has selected Create Account on the product page and does not have a valid account they will be routed to the create account form

- If a user has selected log on from the product page and does not have a valid account they will be routed to the create account form, via an error page.

- If a user has used a direct URL (i.e. has not accessed the product page) the system will behave as if they had selected 'Log In' on the product page and drive the routing appropriately.
 
- A user will be allowed to create COP accounts for each set of authentication credentials e.g. a POISE and digital.homeoffice.gov.uk accounts (existing functionality persists).

Figma UI Acceptance Criteria:
- Added additional views/ pages for the login pages depending on whether the user is active or not active. (follows the Figma logic)

# To test

Given you do not have `COP` account or the `active` flag in Keycloak attribute is set to `no`

- Navigate to the production page
- Click on `Sign in`
- > You should be taken to COP site `/new-account`
- When you type the URL in the browser
- > You should be redirected to `/new-account`
- Create account and submit the form
- > You should be taken to `confirmation-page` where you can `visit the home page`
- > You should see the confirmation panel


![image](https://user-images.githubusercontent.com/30794075/134067040-85b66205-96ab-4b0e-b460-db328d574984.png)
